zk/sequencer/transaction_yielder.go

```diff
@@ -137,17 +139,25 @@ func (y *PoolTransactionYielder) AddMined(hash common.Hash) {
 	// for a transaction to execute.  we still maintain the hash in the toSkip map to avoid yielding it again
 	// when we refresh the pool best list into the readyTransactions slice. there is a window where we have
 	// executed something, but the pool hasn't removed it yet, so we could yield it again before this has happened
+
+	// Search from y.lastYieldIndex backwards
 	foundIdx := -1
-	for idx, readyHash := range y.readyTransactions {
-		if readyHash == hash {
-			foundIdx = idx
+	for i := y.lastYieldIndex; i >= 0; i-- {
+		if y.readyTransactions[i] == hash {
+			foundIdx = i
 			break
 		}
 	}
+
 	if foundIdx != -1 {
-		y.readyTransactions = append(y.readyTransactions[:foundIdx], y.readyTransactions[foundIdx+1:]...)
+		y.readyTransactions = y.readyTransactions[foundIdx+1:]
```

Comment:
could this potentially remove a skipped transaction that we want to attempt in the next run?

Reply:
should be fine. not confirmed with testing